lib/datadog/core/configuration/agent_settings_resolver.rb

```diff
@@ -232,7 +232,30 @@ def uds_fallback
         end
 
         def should_use_uds?
-          can_use_uds? && !mixed_http_and_uds?
+          # When we have mixed settings for http/https and uds, we print a warning
+          # and use the uds settings.
+          mixed_http_and_uds
+          can_use_uds?
+        end
+
+        def mixed_http_and_uds
+          unless defined?(@mixed_http_and_uds)
+            @mixed_http_and_uds = (configured_hostname || configured_port) && can_use_uds?
+            if @mixed_http_and_uds
+              warn_if_configuration_mismatch(
+                [
+                  DetectedConfiguration.new(
+                    friendly_name: 'configuration for unix domain socket',
+                    value: parsed_url.to_s,
+                  ),
+                  DetectedConfiguration.new(
+                    friendly_name: 'configuration of hostname/port for http/https use',
+                    value: "hostname: '#{hostname}', port: '#{port}'",
+                  ),
+                ]
+              )
+            end
+          end
```

Comment:
Don't you mind if I offer to slightly redo the method because it's not returning `@mixed_http_and_uds` and we can spare nesting with guard statements:

```suggestion
          return @mixed_http_and_uds if defined?(@mixed_http_and_uds)
          
          @mixed_http_and_uds = (configured_hostname || configured_port) && can_use_uds?
          if @mixed_http_and_uds
            warn_if_configuration_mismatch(
              [
                DetectedConfiguration.new(
                  friendly_name: 'configuration for unix domain socket',
                  value: parsed_url.to_s,
                ),
                DetectedConfiguration.new(
                  friendly_name: 'configuration of hostname/port for http/https use',
                  value: "hostname: '#{hostname}', port: '#{port}'",
                ),
              ]
            )
          end
          
          @mixed_http_and_uds
```